frontend/src/components/Profile/Profile.tsx

```diff
@@ -32,122 +48,155 @@ const InnerProfile: React.FC = () => {
     setOpen(false);
   };
 
+  useEffect(() => {
+    if (user && user.user_metadata) {
+      setFormData((prev) => ({
+        ...prev,
+        ...user.user_metadata as UserMetadata,
+      }));
+    }
+    if (user && user.email) {
+      setFormData((prev) => ({ ...prev, email: user.email as string }));
+    }
+  }, [user]);
+
+  const handleSave = async () => {
+    if (user) {
+      const { data, error } = await client.auth.updateUser({ data: formData });
+      if (error) {
+        console.error("Error updating user's data:", error);
+      }
+      if (data) {
+        console.log("User updated:", data);
+      }
+    }
+    setIsEditing(false);
+  };
+
   return (
-    <div
-      className="absolute inset-0 h-screen w-screen flex items-center justify-center"
-      style={{ backgroundColor: '#FFCD29' }}
-    >
-      <div className="bg-white p-8 rounded-lg shadow-md w-4/5 h-4/5">
-        <div className="flex h-full">
-          <div className="relative w-1/3 h-1/3 flex flex-col items-center justify-center">
-            <div className="w-40 h-40 rounded-full bg-gray-300 flex items-center justify-center">
-              {image ? (
-                <img
-                  src={image}
-                  alt="Profile"
-                  className="w-40 h-40 rounded-full"
-                />
-              ) : (
-                <AccountCircle style={{ width: 160, height: 160 }} />
-              )}
-            </div>
-            <div className="pt-4"></div>
+    <div className="flex justify-center items-center" style={{ backgroundColor: '#FFCD29', flexGrow: 1, padding: '50px 0' }}>
+      <div className="bg-white p-8 rounded-lg shadow-md w-4/5 h-4/5 flex">
+        <div className="flex flex-col items-center justify-center w-1/3">
+          <div className="w-40 h-40 rounded-full bg-gray-300 flex items-center justify-center mb-6">
+            {image ? (
+              <img src={image} alt="Profile" className="w-40 h-40 rounded-full" />
+            ) : (
+              <AccountCircle style={{ width: 160, height: 160 }} />
+            )}
+          </div>
+          {isEditing && <Button
+            variant="contained"
+            color="primary"
+            className="text-xl mb-6"
+            style={{ padding: '8px 24px' }}
+            onClick={() => setOpen(true)}
+            disabled={!isEditing}
```

Comment:
Is this diabled necessary? It wouldn't show up if isEditing was false right?

Reply:
Upload button will show up when edit is clicked, and you are right, since it wont show up, user cannot click it. Will remove the disable attribute